_posts/2021-11-07-canary.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: "Use a canary in your Apache Kafka coal mine"
+date: 2021-11-08
+author: paolo_patierno
+---
+
+The Strimzi cluster operator is really great to deploy your Kafka cluster on Kubernetes and handling everything related to it.
+It helps with brokers configuration, managing the security aspects, doing the needed upgrades, scaling up and down and many more.
+But ... what if you want to have more insights about how the Kafka cluster is behaving?
+What if you want to know if the Kafka brokers are operating correctly so that clients can exchange messages?
+
+This is where a new component in the Strimzi family comes into the picture ... please welcome to the canary!
+
+<!--more-->
+
+## How a canary can help in a coal mine?
+
+When a Kafka cluster is successfully deployed using the Strimzi operator, the next step is to use it with your own applications or maybe, depending on your business, allowing external users to use it even for integration with your solution.
+But one of the main aspects to consider is also the monitoring and alerting part in order to know if the Kafka cluster is really behaving and working correctly even when you are not really using it a lot.
+Having information about producing and consuming latency, as well as connection times, errors on exchanging messages and so on is really useful for that purpose.
+If your Kafka cluster is exposed to external users you would like to know if they are able to use the cluster itself; the canary can be used to simulate activity for identifying problems from a user perspective even when the cluster seems to be operating correctly.
+Another use case could be when you are handling a fleet of Kafka clusters that you want to update to a newer version doing a canary rolling; just updating a few of them, checking that everything is fine and only then continuing with the rest of the fleet.
+
+These are the main problems that the canary tries to help with.
+Of course, it is not a replacement for any kind of monitoring solution that can be used to show metrics coming from the Kafka cluster, as provided by the corresponding Strimzi examples for setting up Prometheus and Grafana together with useful dashboards.
+
+Like the coal miners using the canaries in the coal mines for early warning about toxic gases in the 20th century, the Strimzi canary can be used to raise alerts when the Kafka cluster is starting to behave badly.
```

Comment:
I wonder if you can find some nice picture or meme to include here ;-). I also wonder if everyone knows the history behind it ... should we maybe add a link to https://en.wikipedia.org/wiki/Sentinel_species or some other web for those who might not understand it?

Reply:
Added the link for now, searching for a good picture or meme ;-)